src/Database/LSMTree/Internal.hs

```diff
@@ -445,25 +445,20 @@ data MergingRun h =
 -- TODO: this should also represent ongoing merges once we implement scheduling.
 newtype MergingRunState h = CompletedMerge (Run h)
 
+-- | Return all runs in the levels, ordered from newest to oldest
+runsInLevels :: Levels h -> V.Vector (Run h)
+runsInLevels levels = flip V.concatMap levels $ \(Level mr rs) ->
+    case mr of
+      SingleRun r                   -> r `V.cons` rs
+      MergingRun (CompletedMerge r) -> r `V.cons` rs
```

Comment:
This could probably be more efficient. Especially when used for `closeLevels`. It might be better to use a traversal function rather than allocating a new vector.

Reply:
Ah! When suggesting to add this function I had in mind that it would return a list, which I would expect to be optimised away when used with either `V.fromList` (for the cache) or `for_` (to close them).